Dada la lista `[4,7, -3]`, calcula su máximo, mínimo y suma.

In [ ]:
lista = [4,7,-3]
print(min(lista))
print(max(lista))
print(sum(lista))


Pon la frase "Born to be wild" en mayúsculas, luego en minúsculas, divídela por espacios y, por último, sustituye "wild" por "Geek".

In [ ]:
"Born to be wild".lower()
"Born to be wild".upper()
"Born to be wild".split()
"Born to be wild".replace("wild", "Geek")

Crea un programa en el que se recojan dos entradas, y la salida del programa sea un booleano que indique al usuario si esas entradas son iguales o no.

In [ ]:
input1 = input("Ingresa un número")
input2 = input("Ingresa un número")
print(input1 == input2)


Imprime los números entre 0 y 70 que son múltiplos de 3 y 5.

In [ ]:
for i in range(0,-22,-7):
    print(i)

In [ ]:
for i in range(-7,-22, -1):
    print(i)

Obtenga la primera y la segunda mejor puntuación de la lista.
La lista contiene duplicados.

In [ ]:
scores = [86,86,85,85,85,83,23,45,84,1,2,0]
set_scores = set(scores)
print(set_scores)
#l_scores = list(set_scores)
l_scores = sorted(set_scores)
print(l_scores[-1], l_scores[-2])
print(l_scores[-2:])

In [ ]:
print(set_scores)
sorted(set_scores)

In [ ]:
scores = [86,86,85,85,85,83,23,45,84,1,2,0]
set_scores = set(scores)
print(set_scores)
l_scores = list(set_scores)
max_scores = max(l_scores)
l_scores.remove(max_scores)
s_max_scores = max(l_scores)
print(max_scores)
print(s_max_scores)

 Haz una lista de diez alienígenas, cada uno de los cuales es de un color: 'rojo', 'verde' o 'azul'.
    - Puedes acortarlo a 'r', 'g' y 'b' si quieres, pero si eliges esta opción tienes que incluir un comentario explicando qué significan r, g y b.
    - Los alienígenas rojos valen 5 puntos, los verdes valen 10 puntos y los azules valen 20 puntos.
    - Utilice un bucle for para determinar el número de puntos que ganaría un jugador por destruir todos los alienígenas de su lista.

In [ ]:
lista = ["rojo", "verde", "rojo", "azul", "azul", "verde", "verde", "azul", "rojo", "azul"]*10_000_000

puntos = 0
for alien in lista:
    if alien == "rojo":
        puntos = puntos + 5 # puntos += 5
    elif alien == "verde":
        puntos = puntos + 10
    else:
        puntos = puntos + 20






In [ ]:
lista = ["rojo", "verde", "rojo", "azul", "azul", "verde", "verde", "azul", "rojo", "azul"]*10_000_000

puntos = 0

rojos = 5*lista.count("rojo")
verde = 10*lista.count("verde")
azul = 20*lista.count("azul")
puntos = rojos + verde + azul
print(puntos)

Crea un programa que calcule la media.

El programa pedirá al usuario que especifique cuántos números quiere introducir, y mostrará un aviso para incluir número a número a una lista, y luego calculará la media de esa lista.

In [ ]:
aviso1 = int(input("Introduzca de cuántos números quieres hacer la media"))

lista_num = []
for i in range(aviso1):
    msg = "Introduzca número " + str(i+1)
    num = float(input(msg))
    lista_num.append(num)

print(sum(lista_num)/aviso1)

print(sum(lista_num)/len(lista_num))


In [ ]:

print(sum(lista_num)/aviso1)

print(sum(lista_num)/len(lista_num))

In [ ]:
len(5)

Haga lo que se describe a continuación:

- Crea una variable llamada fuerza, y fija su valor inicial en 5.

- Imprime un mensaje informando de la fuerza del jugador.

- Establece un bucle while que se ejecute hasta que la fuerza del jugador aumente hasta un valor como 10.

- Dentro del bucle while, imprime un mensaje que informe de la fuerza actual del jugador.

- Dentro del bucle while, escribe una sentencia que incremente la fuerza del jugador.

- Fuera del bucle while, imprime un mensaje informando de que el jugador se ha hecho demasiado fuerte, y que ha pasado a un nuevo nivel del juego.

Juega con diferentes niveles de corte para el valor de la fuerza, y juega con diferentes maneras de aumentar el valor de la fuerza dentro del bucle while.

In [ ]:
fuerza = 5
print("La fuerza del jugardor", fuerza)
while fuerza < 10:
    fuerza += 1  #fuerza = fuerza + 1 
    print("Tu fuerza es de", fuerza)
print("El jugador alcazó un nuevo nivel. Eres demasiado fuerte")

In [ ]:
dni = "0123456789"

if type(dni) != str:
    print("No es una cadena")
if len(dni) > 9:
    print("Es demasiado largo")
if len(dni) < 9:
    print("Es demasiado corto")
if dni[9].isnumeric() == True: ## dni[-1].isalpha() == False
    print("No termina en letra")


In [ ]:
fuerza = 5
print("La fuerza del jugardor", fuerza)
while fuerza < 10:
    fuerza += 1  #fuerza = fuerza + 1 
  

In [ ]:
fuerza = 5
print("La fuerza del jugardor", fuerza)
while fuerza <= 10:
    fuerza += 1  #fuerza = fuerza + 1 
    print("Tu fuerza es de", fuerza)

In [ ]:
fuerza = 5
print("La fuerza del jugardor", fuerza)
while fuerza <= 10:
    print("Tu fuerza es de", fuerza)
    fuerza += 1  #fuerza = fuerza + 1 


Escribe una función que pida un entero e imprima el cuadrado del mismo. Sigue comprobando hasta que la función obtiene un entero.



In [ ]:
# Version 1
def cuadrado(i):
    print(i**2)

entero =  3.2
while (type(entero) != int):
    try:
        entero = int(input("introduzca un entero"))
        cuadrado(entero)
    except:
        print("No es un número")
        



In [ ]:
# Version 2
def cuadrado(i):
    print(i**2)

es_entero =  False
while (es_entero == False):
    try:

        entero = int(input("introduzca un entero"))
        es_entero = True
        cuadrado(entero)

    except:
        print("No es un número")
        continue
    print(entero+1)    


In [ ]:
# Version 3
def cuadrado(i):
    print(i**2)

while (True):
    try:

        entero = int(input("introduzca un entero"))
        cuadrado(entero)
        break

    except:
        print("No es un número")


In [ ]:
# Version 4. Just for fun. No es adecuada ni es necesario conocer
def cuadrado(i):
    print(i**2)
lista = [1]
for i in lista:
    try:

        entero = int(input("introduzca un entero"))
        cuadrado(entero)
        break

    except:
        lista.append(1)
        print("No es un número")

In [ ]:
# Version 5. Just for fun. No es adecuada ni es necesario conocer
def cuadrado():
    try: 
        entero = int(input("introduzca un entero"))
        print(entero**2)
    except: 
        cuadrado()
cuadrado()

Practice_10_OOP Ejercicio 1

In [ ]:
# Para terminar, todo junto, un ejercicio COMPLEJO:
# Escribe un programa que:
# Pida al usuario un número entero por pantalla y continua insistiendo hasta que 
# introduce un número entero correcto
# Si no introduce un número entero, debes imprimir un mensaje de error y volver 
# a pedir otro número entero al usuario
# Si introduce un número distinto de 2, imprimir: "Tu número favorito es X", siendo X el número elegido 
# Si introduce el número 2, escribir "El 2 también es mi numero favorito"

# Cronométrate y envíame lo que tengas tras 30 minutos intentándolo 
# Si me lo envías tras 1 hora itentandolo nunca lo sabré, me engañas a mi pero sobre todo te engañas a ti mismo

# Tras 30 minutos puedes seguir inténtandolo 

In [36]:
while (True):
    try:
        numero = int(input("introduce un número entero por pantalla"))
    except:
        print("NO has introducido un entero")
        continue
    if numero != 2:
        print("Tú numero favorito es", numero)
    else:
        print(f'El {numero} también es mi numero favorito')
    break


NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
Tú numero favorito es 3


In [39]:
condicion = True
while (condicion == True):
    try:
        numero = int(input("introduce un número entero por pantalla"))
    except:
        print("NO has introducido un entero")
        continue
    if numero != 2:
        print("Tú numero favorito es", numero)
    else:
        print(f'El {numero} también es mi numero favorito')
    condicion = False


NO has introducido un entero
NO has introducido un entero
NO has introducido un entero
Tú numero favorito es 45


In [ ]:
# Un último de sucesiones
# Piensa en el ejemplo de Fibonacci
#Las sucesiones pueden ser muy flexibles: partiendo de la lista = [5,7], 
# intenta generar la sucesión que usa la media de los dos números previos y suma uno al resultado
# Sigue pensando en Fibonacci! 

In [7]:
def fibo_sucesion(lista, rango):
    for i in range(rango+1):
        nuevo = (lista[-2]+lista[-1])/2 + 1 
        lista.append(nuevo)
    return lista

l = [5,7]
total = 30
resultado = fibo_sucesion(l.copy(), total)


In [ ]:
def fibo_sucesion(lista, rango):
    for _ in range(rango+1):
        nuevo = (lista[-2]+lista[-1])/2 + 1 
        lista.append(nuevo)
    return lista

l = [5,7]
total = 30
resultado = fibo_sucesion(l.copy(), total)

In [8]:
print(resultado)

[5, 7, 7.0, 8.0, 8.5, 9.25, 9.875, 10.5625, 11.21875, 11.890625, 12.5546875, 13.22265625, 13.888671875, 14.5556640625, 15.22216796875, 15.888916015625, 16.5555419921875, 17.22222900390625, 17.888885498046875, 18.555557250976562, 19.22222137451172, 19.88888931274414, 20.55555534362793, 21.222222328186035, 21.888888835906982, 22.55555558204651, 23.222222208976746, 23.888888895511627, 24.555555552244186, 25.222222223877907, 25.888888888061047, 26.555555555969477, 27.22222222201526]


In [9]:
print(l)

[5, 7, 7.0, 8.0, 8.5, 9.25, 9.875, 10.5625, 11.21875, 11.890625, 12.5546875, 13.22265625, 13.888671875, 14.5556640625, 15.22216796875, 15.888916015625, 16.5555419921875, 17.22222900390625, 17.888885498046875, 18.555557250976562, 19.22222137451172, 19.88888931274414, 20.55555534362793, 21.222222328186035, 21.888888835906982, 22.55555558204651, 23.222222208976746, 23.888888895511627, 24.555555552244186, 25.222222223877907, 25.888888888061047, 26.555555555969477, 27.22222222201526]


In [10]:
def fibo_sucesion(lista, n):
    for i in lista:
        if len(lista) == n:
            break
        nuevo = (lista[-2]+lista[-1])/2 + 1 
        lista.append(nuevo)
    return lista

l = [5,7]
total = 30
resultado = fibo_sucesion(l.copy(), total)


In [11]:
print(resultado)

[5, 7, 7.0, 8.0, 8.5, 9.25, 9.875, 10.5625, 11.21875, 11.890625, 12.5546875, 13.22265625, 13.888671875, 14.5556640625, 15.22216796875, 15.888916015625, 16.5555419921875, 17.22222900390625, 17.888885498046875, 18.555557250976562, 19.22222137451172, 19.88888931274414, 20.55555534362793, 21.222222328186035, 21.888888835906982, 22.55555558204651, 23.222222208976746, 23.888888895511627, 24.555555552244186, 25.222222223877907]


Escribe un programa en Python que genere una lista cuyos valores sean cuadrados de números entre 1 y 30 (ambos incluidos), y devuelva una lista con los 5 primeros y últimos elementos de la lista.

In [12]:
lista_nums = []
for i in range(1,31):
    lista_nums.append(i**2)

lista_5 = lista_nums[0:5] + lista_nums[-5:]
 


In [13]:
print(lista_5)

[1, 4, 9, 16, 25, 676, 729, 784, 841, 900]


In [14]:
lista_ = lista_nums[0:5] + lista_nums[:-5]

In [15]:
print(lista_)

[1, 4, 9, 16, 25, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625]


In [16]:
def funcion_cuadrados():
    lista_nums = []
    for i in range(1,31):
        lista_nums.append(i**2)

    lista_5 = lista_nums[0:5] + lista_nums[-5:]
    return lista_5



In [ ]:
lista_nums = [x**2 for x in range(1,31)]
lista_5_mims = lista_nums[0:5] + lista_nums[-5:]

Escribe un programa que cuente cuantos objetos hay de cada tipo en está lista:


In [18]:
lista = [1, True, 2.3, ['a', 'b'], 'hola', False, False, (2,3,5), 2, 3,5, [2,3,5], "2,3,5", "[2,3,5]"]

In [19]:
dict_tipos = {}
for l in lista:
    if type(l) in dict_tipos.keys():
        dict_tipos[type(l)] += 1
    else:
        dict_tipos[type(l)] = 1          

In [22]:
dict_tipos

{int: 4, bool: 3, float: 1, list: 2, str: 3, tuple: 1}

### Con lista y con diccionario fruta/color

La siguiente lista contiene algunas frutas:
    
`fruits = ['banana', 'orange', 'mango', 'lemon']`

Si una fruta no existe en la lista añade la fruta a la lista e imprime la lista modificada. Si la fruta existe print('Esa fruta ya existe en la lista')

In [28]:
fruits = ['banana', 'orange', 'mango', 'lemon']
fruta = input("Dime una fruta")
if fruta in fruits:
    print('Esa fruta ya existe en la lista')
else:
    fruits.append(fruta)
    print(fruits)

['banana', 'orange', 'mango', 'lemon', 'manzana']


In [32]:
fruits = ['banana', 'orange', 'mango', 'lemon']
fruta = input("Dime una fruta")

if fruta in fruits:
    print('Esa fruta ya existe en la lista')
else:
    try:
        posicion = int(input("En que posición quieres que vaya"))
        fruits.insert(posicion,fruta)
    except:
        fruits.append(fruta)
    print(fruits)




['banana', 'orange', 'mango', 'lemon', 'manzana']


De la lista dada:
    
    a) Crea listas separadas de cadenas y números.

    b) Ordena la lista de cadenas en orden ascendente.

    c) Ordena la lista de cadenas en orden descendente

    d) Ordenar la lista de números de menor a mayor

    e) Ordenar la lista de números de mayor a menor

In [46]:
gadgets = ["Mobile", "Laptop", 100, "Camera", 310.28, "Speakers", 27.00, 
           "Television", 1000, "Laptop Case", "Camera Lens"]
lista_str = []
lista_num = []
for i in gadgets:
    if type(i) == int or type(i) == float:
       lista_num.append(i)
    else:
        lista_str.append(i) 


In [ ]:
gadgets = ["Mobile", "Laptop", 100, "Camera", 310.28, "Speakers", 27.00, 
           "Television", 1000, "Laptop Case", "Camera Lens"]
lista_str = []
lista_num = []
for i in gadgets:
    if type(i) == str:
       lista_str.append(i)
    else:
        lista_num.append(i) 

In [52]:
gadgets = ["Mobile", "Laptop", 100, "Camera", 310.28, "Speakers", 27.00, 
           "Television", 1000, "Laptop Case", "Camera Lens"]
lista_str = []
lista_num = []
for i in gadgets:
    if type(i) != str:
       lista_num.append(i)
    else:
        lista_str.append(i) 

In [54]:
print(lista_num)
print(lista_str)
lista_str_2 = lista_str.copy()

[100, 310.28, 27.0, 1000]
['Mobile', 'Laptop', 'Camera', 'Speakers', 'Television', 'Laptop Case', 'Camera Lens']


In [58]:
lista_str = sorted(lista_str, reverse = False)
print(lista_str)

['Camera', 'Camera Lens', 'Laptop', 'Laptop Case', 'Mobile', 'Speakers', 'Television']


In [57]:
lista_str.sort()
print(lista_str)

['Camera', 'Camera Lens', 'Laptop', 'Laptop Case', 'Mobile', 'Speakers', 'Television']


In [59]:
lista_num.sort()
print(lista_num)

[27.0, 100, 310.28, 1000]


In [60]:
lista_num.sort(reverse=True)
print(lista_num)

[1000, 310.28, 100, 27.0]


Una empresa va a abrir un banco, diseñemos la clase usuario

Debes incluir lo siguiente:

**Un método de retirada**
- Resta dinero del saldo
- Un parámetro, dinero a retirar
- Imprime un mensaje de error por pantalla si no hay saldo para retirar
- Devuelve una cadena con nombre y saldo

**Un método add_cash**
- Añade dinero al saldo
- Un parámetro, dinero a añadir
- Devuelve una cadena con nombre y saldo

**Un método historial**
- LLeva la cuenta de cuantas retiradas ya add_cash se han producido
- Registra toda operación realizada por el usuario: tipo de operación (add_cash, retirada) y el saldo antes y después de la operación 

**Método comisiones**
- Resta 10 euros del saldo al cliente


**Notas adicionales:**

- Todos los usuarios tienen nombre, apellidos y un número de cuenta que siempre será distinto
- Ningún número de entrada será negativo



In [91]:
class Usuario():

 
    def __init__(self, nombre, apellido1, apellido2, n_cuenta, saldo = 0):

        self.nombre = nombre
        self.primer_apellido = apellido1
        self.segundo_apellido = apellido2
        self.num_cuenta = n_cuenta
        self.saldo = saldo
        self.registro = []
      

    def retirada(self, dinero):
        self.historial( -dinero)
        if dinero > self.saldo:
            return "La cantidad a retirar es mayor que el saldo"
        else:
            self.saldo -= dinero # self.saldo = self.saldo - dinero
            return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
        
    def add_cash(self, dinero):
        self.historial(dinero)
        self.saldo += dinero #self.saldo = self.saldo + dinero
        return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
    
    def comisiones(self):
        self.historial( -10)
        self.saldo = self.saldo - 10

    def historial(self, dinero):
        
        if dinero >= 0:
            tipo = "add_cash"
        else:
            tipo = "retirada"
        oper = {"saldo_antes": self.saldo , "saldo_despues": self.saldo + dinero, "tipo": tipo}
        self.registro.append(oper)        
        


        
    

In [ ]:
class Usuario():

 
    def __init__(self, nombre, apellido1, apellido2, n_cuenta, saldo = 0):

        self.nombre = nombre
        self.primer_apellido = apellido1
        self.segundo_apellido = apellido2
        self.num_cuenta = n_cuenta
        self.saldo = saldo
        self.registro = []
      

    def retirada(self, dinero):
        saldo_antes = self.saldo
        if dinero > self.saldo:
            return "La cantidad a retirar es mayor que el saldo"
        else:
            self.saldo -= dinero # self.saldo = self.saldo - dinero
            saldo_despues = self.saldo
            self.historial("retirada", saldo_antes, saldo_despues)
            return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
        
    def add_cash(self, dinero):
        saldo_antes = self.saldo
    
        self.saldo += dinero #self.saldo = self.saldo + dinero
        saldo_despues = self.saldo
        self.historial("add_cash", saldo_antes, saldo_despues)
        return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
    
    def comisiones(self):
        saldo_antes = self.saldo
        self.saldo = self.saldo - 10
        saldo_despues = self.saldo
        self.historial("comisiones", saldo_antes, saldo_despues)

    def historial(self, operation, saldo_antes, saldo_despues):
        new_oper = {"saldo_antes": saldo_antes , "saldo_despues": saldo_despues, "tipo": operation}
        self.registro.append(new_oper)        
        

        
           
        

In [94]:
class Usuario():

 
    def __init__(self, nombre, apellido1, apellido2, n_cuenta, saldo = 0):

        self.nombre = nombre
        self.primer_apellido = apellido1
        self.segundo_apellido = apellido2
        self.num_cuenta = n_cuenta
        self.saldo = saldo
        self.registro = []
      

    def retirada(self, dinero):
        saldo_antes = self.saldo
        if dinero > self.saldo:
            return "La cantidad a retirar es mayor que el saldo"
        else:
            self.saldo -= dinero # self.saldo = self.saldo - dinero
            saldo_despues = self.saldo
            self.registro.append({"saldo_antes": saldo_antes , "saldo_despues": saldo_despues, "tipo": "retirada"})
            return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
        
    def add_cash(self, dinero):
        saldo_antes = self.saldo
        self.saldo += dinero #self.saldo = self.saldo + dinero
        saldo_despues = self.saldo
        self.registro.append({"saldo_antes": saldo_antes , "saldo_despues": saldo_despues, "tipo": "add_cash"})
        return f'{self.nombre}:{self.saldo}' # self.nombre + ":" + self.saldo
    
    def comisiones(self):
        saldo_antes = self.saldo
        self.saldo = self.saldo - 10
        saldo_despues = self.saldo
        self.registro.append({"saldo_antes": saldo_antes , "saldo_despues": saldo_despues, "tipo": "comisiones"})
     

    

In [95]:
user1 = Usuario("Jon", "Doe", "Smith", 123)
user1.add_cash(25)

'Jon:25'

In [96]:
user1.registro

[{'saldo_antes': 0, 'saldo_despues': 25, 'tipo': 'add_cash'}]

Ahora la clase Banco:

**Método añadir cliente**
Añade un nuevo cliente al banco. (lista, diccionario, tupla?)
No pueden existir dos números de cuenta iguales: se sobreescribe dicho cliente

**Método eliminar cliente**
Se elimina un cliente si existe. 
Si no existe imprimer mensaje de que no existe dicho cliente

**Saldo total del banco**
Devuelve la totalidad del saldo sumado para todos los clientes

**Metodo fin de mes**
Cobra la comisión mensual a los clientes
Calcula las comisiones totales cobradas
Calcula el saldo total de los clientes del banco tras el pago de comisiones


In [ ]:
# Crea un programa que pida al usuario números para calcular la raíz cuadrada de los mismos. Captura la excepción de que haya introducido un número negativo 
# y haz que calcule la raíz cuadrada del valor absoluto en esa situación
# El programa continua pidiendo al usuario números y calculando la raíz hasta que el usuario introduzce el número 0, en cuyo caso, termina el programa.

In [ ]:
# Escribe un programa que pida palabras al usuario. 
# Si la palabra está en la frase "La lluvia en Sevilla es una maravilla", imprimirá "Es un acierto" 
# Si la palabra no está en dicha frase, imprimirá "Es un fallo" y restará una vida
# El número total de vidas es 4
# EL programa no es sensible a mayúsculas ni minúsuclas

Dada una lista con 3 números del 1 al 20, escribe un programar que pida al usuario 5 números del 1 al 20 y lleve la cuenta del total de aciertos y fallos y los números introducidos.
Si se aciertan los 3 números antes de finalizar las 5 rondas el programa debe parar
Si no se han acertado los 3 números, al final deben imprimirse los números que no fueron acertados.


In [98]:
lista_aletoria = [3, 8, 16]

acierto = 0
fallo = 0
numeros_introducidos = []
for i in range(5):

    num = int(input("Introduce un número del 1 al 20"))

    if num in lista_aletoria:
        acierto = acierto + 1
    else:
        fallo = fallo + 1
    numeros_introducidos.append(num)
    if acierto == 3:
        print("Has ganado")
        break

print(set(lista_aletoria) - set(numeros_introducidos))
    


{8, 16}


In [99]:
[x for x in lista_aletoria if x not in numeros_introducidos]

[8, 16]

Los ejercicios hechos sin OOP, usar OOP y los de OOP, pasar a funciones

# Ramp up 2022

### Ejercicio 1
En este ejercicio se va a crear una lista con los siguientes elementos. Primero, crearemos las variables que luego añadiremos a la lista. Las variables son las siguientes:

1. Crea una variable llamada `num_dni` de tipo Integer que represente el primer número de tu DNI. 
2. Crea una variable llamada `let_dni` de tipo String y que represente la letra de vuestro DNI.
3. Crea una variable llamada `tip_ord` de tipo Boolean indicando si usas Windows o no.
4. Crea una variable llamada `nada` de tipo None cuyo valor sea None.
5. Crea una variable llamada `lista_deseo` de tipo List que contenga los cuatro objetos que te llevarías a una isla desierta. Cada elemento ha de ser de tipo String y el tamaño de la lista ha de ser 4.
6. Crea una variable llamada `tupla_ropa` de tipo Tuple con la marca de la prenda superior, inferior y calzado que llevas puesto.
7. Crea una variable llamada `dict_info` con dos claves y dos valores. Una clave es 'nombre', teniendo como valor vuestro nombre. Otra key es 'edad', teniendo como valor vuestra edad.
8. Crea una variable llamada `flo_hor` con la hora actual separada por punto en vez de dos puntos. Ejemplo: si son las 10:06, el valor de la variable sería 10.06

Añade todos estos elementos en una lista llamada `lista_examen` en el orden de aparición:
```Python
lista_examen = [num_dni, let_dni, ...]
```

### Ejercicio 2

Para resolver este ejercicio necesitarás la lista del ejercicio anterior (`lista_examen`). Se recomienda leer cada punto hasta el final antes de resolver el ejercicio.

Se repite para que quede aún más claro: **SOLO USAR LA VARIABLE** `lista_examen`. Usar las demás variables se calificará como error.  

1. Mostrar por pantalla, sin uso de funciones ni bucles: 
    - El segundo elemento de `lista_deseo` desde `lista_examen`.
    - El calzado que llevas puesto.
    - Vuestro nombre.
    - La hora almacenada concatenada con vuestra edad y el número de vuestro de DNI. No debe haber espacios entre la concatenación.
    - Si tenéis más de 20 años, el primer número de vuestro DNI es mayor a 7 y usas Mac, mostrar la posición 5 de `lista_examen`. En caso contrario, imprime el mensaje 'No se cumple'

2. Mostrar por pantalla, con uso de un bucle `for`:
    - Todos los elementos de `lista_examen`.
    - Todos los elementos de `lista_deseo` desde `lista_examen`.
    - Todos los elementos que ocupen una posición menor a 4 en `lista_examen`.

### Ejercicio 3 
Implementa una función que reciba una lista como argumento, calcule la raíz cuadrada de cada uno de los elementos de la lista y devuelva en un diccionario el valor máximo y mínimo de la lista de raíces. 

Ejemplo:
```Python
entrada: [4,9,36]
salida: {'max':6.0,'min':2.0}
```

### Ejercicio 4

Crea una clase "Jugador" que instancie objetos con 3 atributos: "goles", "dorsal" y "nombre". Además, debe contar con dos métodos: 

- `marcar_gol`: al llamar a este método se debe sumar 1 a los goles actuales del jugador
- `cambiar_dorsal` que recibe parametro "nuevo_dorsal": al llamar a esta función con un parámetro de entrada, esta función debe cambiar el número del dorsal por el "nuevo_dorsal"

Crea un jugador llamado "Rafinha" que lleva 0 goles y usa el dorsal 99. A Rafinha nunca le gustó su dorsal. Por eso, usa la función correspondiente para cambiarle su dorsal al 13. Además, en el último partido marcó un golazo: utiliza la función para aumentar sus goles anotados. 

Muestra por pantalla el dorsal del jugador antes y después de cambiarse de dorsal. Muestra también los goles de Rafinha antes y después de haber marcado.

# Ramp up 2022

Ejercicio 1 (0.5 punto)

Escribe un programa que pida al usuario dos números y devuelva la suma de los mismos, evitando errores.


Ejercicio 2 (0.5 puntos)

Escribe una función que reciba una lista de números y devuelva la suma de los elementos pares de la lista.


Ejercicio 3 (0.5 puntos)

Escribe un programa que lea una tupla de números enteros y devuelva una lista con los mismos números en orden inverso.

Ejercicio 4 (0.5 puntos)

Escribe una función que reciba una lista de números y retorne un diccionario con el mínimo, el máximo, la media y la mediana de esos números. sin utilizar librerias externas

Ejercicio 5 (0.5 puntos)

Escribe un texto en formato Markdown que incluya al menos un encabezado, una lista numerada y una imagen.

Ejercicio 6 (2.5 puntos)

Crea una clase llamada Saiyajin que tenga los siguientes métodos y atributos:

Atributos para el nombre, nivel de poder y si es Super Saiyajin o no.

 kamehameha(): calcula el nivel de poder antes del ataque multiplicado por diez, acto seguido disminuye el nivel de poder antes del ataque en un 15% y devuelve una cadena que dice "Kamehameha por X" siendo X el valor del nivel de poder antes del ataque por diez . 
 
 transformar(): Cambia el estado de Super Saiyajin y ajusta el nivel de poder en consecuencia. Si no estaba en SUper Saiyajin pasa a estarlo e incrementa el poder en un 300%. Si estaba en Super Saiyajin pasa a no estarlo y divide entre 3 el poder.

 Crea una lista de al menos 6 objetos de tipo Saiyajin y determina el de mayor nivel de poder, el de menor nivel de poder, el nivel medio de poder y cuantos super saiyajin hay en la lista

 Crea una lista con los niveles de poder y ordenala de menor a mayor.

Ejercicio 7 (2 puntos)

Escribe un programa que realice funciones basicas de una calculadora.
Suma, resta, multiplicación y división
Debe preguntar al usuario la operación a realizar y los operandos
Debe gestionar errores: operación o números no válidos y división entre 0.